accounts/accounts.go

```diff
@@ -132,3 +140,28 @@ func (ac *Account) RemoveFunds(amount currency.Currency) error {
 	}
 	return nil
 }
+
+// PopLatestBuy : Returns and removes the most recent Buy in a queue
+func (ac *Account) PopLatestBuy() (Action, bool) {
+	// gobuild says:
+	//   Can't use Action as nil so we send back a bool to indicate hit/miss
+
+	// Check for empty queue
+	queueLen := len(ac.BuyQueue)
+	if queueLen == 0 {
+		return Action{}, false
+	}
+
+	// Last item appended to queue will be the most recent.
+	// Copy the last item then shrink the queue.
+	var latestAction Action
+	latestAction, ac.BuyQueue = ac.BuyQueue[queueLen-1], ac.BuyQueue[:queueLen-1]
```

Comment:
Go doesn't have list comprehensions built in so this is the idiomatic way to pop something 😭 

https://github.com/golang/go/wiki/SliceTricks